.travis/prepare.sh

```diff
@@ -1,15 +1,33 @@
 #!/bin/bash
 
-CLANG_DIR="clang+llvm-3.8.1-x86_64-linux-gnu-ubuntu-14.04"
-CLANG_FILE="${CLANG_DIR}.tar.xz"
-CLANG_URL="http://releases.llvm.org/3.8.1/$CLANG_FILE"
+CLANG_VERSION=3.8.1
 
-wget -nv $CLANG_URL
-sudo rm -rf /usr/local/clang
-sudo mkdir -p /usr/local
-sudo tar -C /usr/local -xJf $CLANG_FILE
-sudo ln -s /usr/local/$CLANG_DIR /usr/local/clang
-rm $CLANG_FILE
+function setup_env() {
+case `uname -m` in
+  'x86_64' )
+    CLANG_DIR="clang+llvm-$CLANG_VERSION-x86_64-linux-gnu-ubuntu-14.04"
+    ;;
+  'aarch64' )
+    CLANG_DIR="clang+llvm-$CLANG_VERSION-aarch64-linux-gnu"
+    sed -i 's/etcd:v3.2.17/etcd:v3.2.17-arm64/g' $GOPATH/src/github.com/cilium/cilium/Makefile.defs
```

Comment:
instead of using sed, can't you run make with `ETCD_IMAGE` set depending on the architecture?

Reply:
> instead of using sed, can't you run make with `ETCD_IMAGE` set depending on the architecture?

Conditional statements can be added to Makefile.defs, for example:
+ifeq ($(shell uname -m),aarch64)
+    ETCD_IMAGE=quay.io/coreos/etcd:v3.2.17-arm64
+else
+    ETCD_IMAGE=quay.io/coreos/etcd:v3.2.17
+endif

How about you  opinion on this modification?

Reply:
I was thinking on having it in `./.travis/build.sh` but the conditional statement in the Makefile.defs sounds good to me as well.

Reply:
Done